content/tokio/tutorial/non-async.md

```diff
@@ -0,0 +1,233 @@
+---
+title: "Non-async projects"
+---
+
+In the examples we have seen so far, we marked the main function with
+`#[tokio::main]` and made the entire project asynchronous. However you don't
+have to do this to use Tokio in your project. This page explains how you can
+isolate async/await to a small part of your project.
+
+# What `#[tokio::main]` expands to
+
+The `#[tokio::main]` macro is a macro that replaces your main function with a
+non-async main function that starts a runtime and then calls your code. For
+instance, this:
+```rust
+#[tokio::main]
+async fn main() {
+    println!("Hello world");
+}
+```
+is turned into this:
+```
+fn main() {
+    tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(async {
+            println!("Hello world");
+        })
+}
+```
+by the macro. To use async/await in our own projects, we will do something
+similar.
+
+# Ways to approach it
+
+There are a few ways you can approach the problem of using async/await in only
+part of your project. These are:
+
+ * Create a [`Runtime`] and call [`block_on`] on the async code.
+ * Create a [`Runtime`] and [`spawn`] things on it.
+ * Run the [`Runtime`] in a separate thread and send messages to it.
+
+The main differences between the above approaches are whether the runtime is
+able to run stuff in the background, and how you shut the runtime down.
+
+## Using `block_on`
+
+The simplest way to transition from synchronous to asynchronous code is to call
+the [`block_on`] method. To do this, you first create a [`Runtime`], then call
+[`block_on`]. For example:
+```rust
+use tokio::runtime::Builder;
+
+fn main() {
+    println!("In non-async code!");
+
+    let runtime = Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap();
+
+    runtime.block_on(my_async_fn());
+
+    println!("Back in non-async code!");
+}
+
+async fn my_async_fn() {
+    println!("Now in async code!");
+}
+```
+If you compare this to the previous example from the `#[tokio::main]` macro you
+will see that it is very similar, however there is one difference: This example
+uses the [`current_thread`] runtime rather than the [`multi_thread`] runtime.
+Either choice would work, but the `current_thread` runtime is cheaper to use
+because it does not spawn any threads.
+
+Be aware that using the `current_thread` runtime has the consequence that, since
```

Comment:
I think the "warning" style would work for this paragraph. This page has an example of the syntax, but I can't link to it for some reason: https://github.com/tokio-rs/website/blob/master/content/tokio/tutorial/hello-tokio.md